index.js

```diff
@@ -37,41 +38,163 @@ var EchoClient = function(config){
-        throw new Error('Missing field data.class');
+
+    if (!queryParams) {
+        throw new Error('Missing Analytics queryParams');
     }
-    if(!data.source){
-        throw new Error('Missing field data.source');
+
+    var constructQueryStringResponse = this._queryStringParams(queryParams);
+
+    if (constructQueryStringResponse.errors) {
+        throw new Error('Invalid Analytics queryParams');
     }
 
     var requestOptions = {
-        url: this.config.echo_endpoint+'/1/events',
+        url: this.config.echo_endpoint + '/1/analytics/' + queryOperator + '?' + constructQueryStringResponse.queryString,
+        method: 'GET',
         headers: {
-            'Accept': 'application/json',
-            'Authorization':'Bearer '+token
-        },
-        body:data,
-        method:'POST',
-        json:true
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer ' + token
+        }
     };
 
-    request.post(requestOptions, function(err, response, body){
-        if(err){
+    if (useCache === false) {
+        requestOptions.headers['cache-control'] = 'none';
+    }
+
+    this.debug(JSON.stringify(requestOptions));
+
+    request.get(requestOptions, function(err, response, body) {
+        if (err) {
             callback(err);
-        } else{
+        } else {
             callback(null, body);
         }
     });
+};
 
-    this.debug(JSON.stringify(requestOptions));
+/**
+ * Build up a query string
+ * @param {object} params
+ * @returns {string}
+ * @private
+ */
+EchoClient.prototype._queryStringParams = function(params) {
+    var queryString = '';
+    var queryStringParams = [];
+    var paramErrors = null;
+
+    var isValidParameter = function (parameter) {
+        var validParamters = [
+            'class',
+            'source',
+            'property',
+            'interval',
+            'group_by',
+            'key',
+            'value',
+            'from',
+            'to',
+            'percentile',
+            'user',
+            'filter',
+            'n'
+        ];
 
+        var hasDot = parameter.indexOf('.');
+
+        if (hasDot > -1) {
+            parameter = parameter.substring(0, hasDot);
+        }
+
+        if (validParamters.indexOf(parameter) === -1) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    if (!_.isEmpty(params)) {
+        for (var i in params) {
+            if (params.hasOwnProperty(i)) {
+                if (isValidParameter(i)) {
+                    queryStringParams.push(encodeURIComponent(i) + '=' + encodeURIComponent(params[i]));
+                } else {
+                    if (!paramErrors) {
+                        paramErrors = [];
+                    }
+                    paramErrors.push(i);
+                }
+            }
+        }
+        queryString += queryStringParams.join('&');
+    }
+
+    return { errors: paramErrors, queryString: queryString };
```

Comment:
Why not just immediately throw an error on the first encountered param error instead of returning a bunch of them?